pcsx2/Input/SDLInputSource.cpp

```diff
@@ -21,92 +21,92 @@
+	{ICON_PF_LEFT_ANALOG_LEFT, ICON_PF_LEFT_ANALOG_RIGHT}, // SDL_GAMEPAD_AXIS_LEFTX
+	{ICON_PF_LEFT_ANALOG_UP, ICON_PF_LEFT_ANALOG_DOWN}, // SDL_GAMEPAD_AXIS_LEFTY
+	{ICON_PF_RIGHT_ANALOG_LEFT, ICON_PF_RIGHT_ANALOG_RIGHT}, // SDL_GAMEPAD_AXIS_RIGHTX
+	{ICON_PF_RIGHT_ANALOG_UP, ICON_PF_RIGHT_ANALOG_DOWN}, // SDL_GAMEPAD_AXIS_RIGHTY
+	{nullptr, ICON_PF_LEFT_TRIGGER_PULL}, // SDL_GAMEPAD_AXIS_LEFT_TRIGGER
+	{nullptr, ICON_PF_RIGHT_TRIGGER_PULL}, // SDL_GAMEPAD_AXIS_RIGHT_TRIGGER
 };
 static constexpr const GenericInputBinding s_sdl_generic_binding_axis_mapping[][2] = {
-	{GenericInputBinding::LeftStickLeft, GenericInputBinding::LeftStickRight}, // SDL_CONTROLLER_AXIS_LEFTX
-	{GenericInputBinding::LeftStickUp, GenericInputBinding::LeftStickDown}, // SDL_CONTROLLER_AXIS_LEFTY
-	{GenericInputBinding::RightStickLeft, GenericInputBinding::RightStickRight}, // SDL_CONTROLLER_AXIS_RIGHTX
-	{GenericInputBinding::RightStickUp, GenericInputBinding::RightStickDown}, // SDL_CONTROLLER_AXIS_RIGHTY
-	{GenericInputBinding::Unknown, GenericInputBinding::L2}, // SDL_CONTROLLER_AXIS_TRIGGERLEFT
-	{GenericInputBinding::Unknown, GenericInputBinding::R2}, // SDL_CONTROLLER_AXIS_TRIGGERRIGHT
+	{GenericInputBinding::LeftStickLeft, GenericInputBinding::LeftStickRight}, // SDL_GAMEPAD_AXIS_LEFTX
+	{GenericInputBinding::LeftStickUp, GenericInputBinding::LeftStickDown}, // SDL_GAMEPAD_AXIS_LEFTY
+	{GenericInputBinding::RightStickLeft, GenericInputBinding::RightStickRight}, // SDL_GAMEPAD_AXIS_RIGHTX
+	{GenericInputBinding::RightStickUp, GenericInputBinding::RightStickDown}, // SDL_GAMEPAD_AXIS_RIGHTY
+	{GenericInputBinding::Unknown, GenericInputBinding::L2}, // SDL_GAMEPAD_AXIS_LEFT_TRIGGER
+	{GenericInputBinding::Unknown, GenericInputBinding::R2}, // SDL_GAMEPAD_AXIS_RIGHT_TRIGGER
 };
 
 static constexpr const char* s_sdl_button_names[] = {
-	"A", // SDL_CONTROLLER_BUTTON_A
-	"B", // SDL_CONTROLLER_BUTTON_B
-	"X", // SDL_CONTROLLER_BUTTON_X
-	"Y", // SDL_CONTROLLER_BUTTON_Y
-	"Back", // SDL_CONTROLLER_BUTTON_BACK
-	"Guide", // SDL_CONTROLLER_BUTTON_GUIDE
-	"Start", // SDL_CONTROLLER_BUTTON_START
-	"LeftStick", // SDL_CONTROLLER_BUTTON_LEFTSTICK
-	"RightStick", // SDL_CONTROLLER_BUTTON_RIGHTSTICK
-	"LeftShoulder", // SDL_CONTROLLER_BUTTON_LEFTSHOULDER
-	"RightShoulder", // SDL_CONTROLLER_BUTTON_RIGHTSHOULDER
-	"DPadUp", // SDL_CONTROLLER_BUTTON_DPAD_UP
-	"DPadDown", // SDL_CONTROLLER_BUTTON_DPAD_DOWN
-	"DPadLeft", // SDL_CONTROLLER_BUTTON_DPAD_LEFT
-	"DPadRight", // SDL_CONTROLLER_BUTTON_DPAD_RIGHT
-	"Misc1", // SDL_CONTROLLER_BUTTON_MISC1
-	"Paddle1", // SDL_CONTROLLER_BUTTON_PADDLE1
-	"Paddle2", // SDL_CONTROLLER_BUTTON_PADDLE2
-	"Paddle3", // SDL_CONTROLLER_BUTTON_PADDLE3
-	"Paddle4", // SDL_CONTROLLER_BUTTON_PADDLE4
-	"Touchpad", // SDL_CONTROLLER_BUTTON_TOUCHPAD
+	"A", // SDL_GAMEPAD_BUTTON_SOUTH
+	"B", // SDL_GAMEPAD_BUTTON_EAST
+	"X", // SDL_GAMEPAD_BUTTON_WEST
+	"Y", // SDL_GAMEPAD_BUTTON_NORTH
```

Comment:
These buttons will now be wrong on Nintendo controllers, might want to rename them and add a backwards compatibility thing when parsing config files.

We also have to deal with FSUI controls with OK/Cancel being swapped

Reply:
Where the Nintendo buttons correct before?

Reply:
The SDL2 handling would be to make "A" the A button, ignoring its physical location on the controller
So the auto mapping would be incorrect, but everything else (e.g. the labels displayed in the binding UI, and which is OK/Cancel in FSUI) would be correct.

Now the auto mapping is correct, and everything else is incorrect.

Reply:
That would mean those with Nintendo controllers would need to redo their binds then
Although we probably still want backwards comparability for other mapped gamepads?

Reply:
Correct.  So maybe the solution is to get everyone with legacy ABXY mappings back into the mapping interface to rebind their controllers.  Unless you want to make a fancy system that supports both ABXY and NESW mappings, someone's mapping files are going to break with this upgrade, and we probably have to accept that.  I guess we could also check if a user is both using ABXY mappings and using a controller where `SDL_GetGamepadButtonLabel(SDL_GAMEPAD_BUTTON_SOUTH) == SDL_GAMEPAD_BUTTON_LABEL_B` and issue a prompt then to rebind the controller.

Reply:
Legacy ABXY mappings are remapped based on the connected gamepad's layout, and then saved with that new mapping.
A message prompt is presented if the face button layout differs from an Xbox controller (and also isn't a PS controller)

The FSUI binds being swapped can be dealt with later, We should probably just have the option to swap OK/Cancel regardless of controller